src/store/models/designer.ts

```diff
@@ -213,6 +219,43 @@ const designerModel: DesignerModel = {
       }
     });
   }),
+  renameNode: thunk(async (actions, { nodeId, name }, { getState, getStoreActions }) => {
+    const { allCharts, activeId } = getState();
+    const chart = allCharts[activeId];
+
+    // Ensure the node exists
+    const node = chart.nodes[nodeId];
+    if (!node) {
+      throw new Error(`${node} node with id ${nodeId} not found.`);
```

Comment:
`${node}` will always be `undefined` here so there's no reason to include it in the error message.